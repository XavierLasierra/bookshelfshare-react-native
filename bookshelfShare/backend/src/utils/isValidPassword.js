const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
