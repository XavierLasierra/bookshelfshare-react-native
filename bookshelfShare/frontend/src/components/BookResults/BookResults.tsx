import React, { useEffect } from 'react';
import {
  SafeAreaView, ScrollView, Text, ActivityIndicator, FlatList
} from 'react-native';

import { useDispatch, useSelector } from 'react-redux';
import Header from '../Header/Header';

import SearchIcon from '../../assets/searchIcon.svg';
import { clearBooks } from '../../redux/actions/books.creator';
import stylesConstants from '../../styles/styles.constants';
import BookElementSearch from '../BookElementSearch/BookElementSearch';

interface Props {
  route: Route,
  navigation: any
}
interface Route {
  params: Params

}
interface Params {
  searchInformation: SearchInformation
}
interface SearchInformation {
  isbn: string,
  inauthor: string,
  intitle: string,
  inpublisher: string,
}

export default function BookResults(
  {
    navigation, route: {
      params: {
        searchInformation
      }
    }
  }: Props
) {
  const dispatch = useDispatch();
  const { books, results } = useSelector((store: any) => store.books);
  const {
    isbn, inauthor, intitle, inpublisher
  } = searchInformation;

  useEffect(() => () => {
    dispatch(clearBooks());
  }, []);

  function renderBook({ item }: any) {
    return <BookElementSearch bookData={item} />;
  }
  return (
    <SafeAreaView>
      <Header Logo={SearchIcon} BackButton navigation={navigation} />
      <ScrollView>
        <Text>
          {isbn && `ISBN: ${isbn}`}
          {inauthor && `Author: ${inauthor}`}
          {intitle && `Title: ${intitle}`}
          {inpublisher && `Publisher: ${inpublisher}`}
        </Text>
        {results
          ? (
            <>
              {books.length > 0
                ? (
                  <FlatList
                    data={books}
                    renderItem={renderBook}
                    keyExtractor={(book) => book?.isbn[0]?.identifier}
                  />
                )
                : <Text>0 results</Text>}
            </>
          )
          : <ActivityIndicator size="large" color={stylesConstants.colors.dark} />}
      </ScrollView>
    </SafeAreaView>
  );
}
