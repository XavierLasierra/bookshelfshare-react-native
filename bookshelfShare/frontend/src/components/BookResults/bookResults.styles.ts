import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  searchParameters: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    paddingVertical: 10
  },
  parameter: {
    fontSize: stylesConstants.fontSize.small,
    color: stylesConstants.colors.mainText,
    paddingHorizontal: 5
  },
  bookResultsContainer: {
    backgroundColor: stylesConstants.colors.light
  },
  resultsContainer: {
    alignItems: 'center',
    height: '100%'
  },
  marginBottom: {
    width: '100%',
    height: 130
  }
});
