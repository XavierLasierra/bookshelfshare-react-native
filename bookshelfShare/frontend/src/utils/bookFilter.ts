export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
