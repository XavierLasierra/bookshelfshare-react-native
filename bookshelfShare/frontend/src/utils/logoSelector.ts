import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
