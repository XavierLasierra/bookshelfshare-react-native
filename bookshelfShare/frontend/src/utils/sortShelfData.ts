export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
