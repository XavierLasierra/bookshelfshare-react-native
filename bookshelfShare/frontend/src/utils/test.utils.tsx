import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
