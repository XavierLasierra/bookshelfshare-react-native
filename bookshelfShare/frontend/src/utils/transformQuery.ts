interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
