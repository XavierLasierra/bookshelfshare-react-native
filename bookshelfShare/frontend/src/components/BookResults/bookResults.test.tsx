import React from 'react';
import BookResults from './BookResults';
import { render } from '../../utils/test.utils';
import bookDetailsMock from '../../mocks/bookDetails.mock';

const navigation = {};

describe('Given a BookResults component', () => {
  describe('When it is rendered', () => {
    describe('And book results are loading searching by isbn', () => {
      const route = {
        params: {
          searchInformation: {
            isbn: '1',
            inauthor: '',
            intitle: '',
            inpublisher: ''
          }
        }
      };
      test('Then should match the snapshot', () => {
        const screen = render(<BookResults navigation={navigation} route={route} />);
        expect(screen).toMatchSnapshot();
      });
    });
    describe('And there are no book results searching by author and publisher', () => {
      const route = {
        params: {
          searchInformation: {
            isbn: '',
            inauthor: 'author',
            intitle: '',
            inpublisher: 'publisher'
          }
        }
      };
      test('Then should match the snapshot', () => {
        const initialState = {
          books: {
            books: [],
            results: true
          }
        };
        const screen = render(<BookResults navigation={navigation} route={route} />, initialState);
        expect(screen).toMatchSnapshot();
      });
    });

    describe('And there are book results searching by author, title and publisher', () => {
      const route = {
        params: {
          searchInformation: {
            isbn: '',
            inauthor: 'author',
            intitle: 'title',
            inpublisher: 'publisher'
          }
        }
      };
      test('Then should match the snapshot', () => {
        const initialState = {
          books: {
            books: bookDetailsMock,
            results: true
          }
        };
        const screen = render(<BookResults navigation={navigation} route={route} />, initialState);
        expect(screen).toMatchSnapshot();
      });
    });
  });
});
