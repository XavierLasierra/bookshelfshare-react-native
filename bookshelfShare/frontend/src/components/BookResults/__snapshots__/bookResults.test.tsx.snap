// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BookResults component When it is rendered And book results are loading searching by isbn Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        ISBN: 1
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        
      </Text>
    </View>
    <ActivityIndicator
      color="#000000"
      size="large"
    />
    <View
      style={
        Object {
          "height": 130,
          "width": "100%",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`Given a BookResults component When it is rendered And there are book results searching by author, title and publisher Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        Title: TITLE
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        Author: AUTHOR
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        Publisher: PUBLISHER
      </Text>
    </View>
    <RCTScrollView
      data={
        Array [
          Object {
            "authors": Array [
              "George Orwell",
            ],
            "categories": Array [
              "London (England)",
            ],
            "description": "Winston Smith is a worker at the Ministry of Truth, where he falsifies records for the party. Secretly subversive, he and his colleague Julia try to free themselves from political slavery but the price of freedom is betrayal.",
            "format": "BOOK",
            "images": Object {
              "smallThumbnail": "smallThumbnail1",
              "thumbnail": "thumbnail1",
            },
            "isbn": Object {
              "ISBN10": "0451524934",
              "ISBN13": "9780451524935",
            },
            "language": "en",
            "pageCount": 190,
            "publishedDate": "2009",
            "publisher": "No publisher found",
            "subtitle": "A Novel",
            "title": "1984",
          },
          Object {
            "authors": Array [
              "George Orwell",
              "Another author",
              "Last author",
            ],
            "categories": Array [
              "London (England)",
              "Fiction",
              "Another",
            ],
            "description": "Winston Smith is a worker at the Ministry of Truth, where he falsifies records for the party. Secretly subversive, he and his colleague Julia try to free themselves from political slavery but the price of freedom is betrayal.",
            "format": "BOOK",
            "images": Object {
              "smallThumbnail": "smallThumbnail2",
            },
            "isbn": Object {
              "ISBN10": "0451524934",
              "ISBN13": "9780451524935",
            },
            "language": "en",
            "pageCount": 190,
            "publishedDate": "2009",
            "publisher": "No publisher found",
            "subtitle": "A Novel",
            "title": "1984",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginVertical": 10,
                "opacity": 1,
              }
            }
            testID="bookDetailButton"
          >
            <Image
              source={
                Object {
                  "uri": "thumbnail1",
                }
              }
              style={
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderTopLeftRadius": 10,
                  "height": 200,
                  "overflow": "hidden",
                  "width": 125,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 175,
                  "justifyContent": "space-between",
                  "padding": 10,
                  "width": 210,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "maxHeight": 75,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "width": 175,
                      }
                    }
                  >
                    1984
                  </Text>
                  <Text>
                    EN
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                    }
                  }
                >
                  George Orwell
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3E3C37",
                      "fontSize": 12,
                    }
                  }
                >
                  No publisher found
                </Text>
              </View>
              <View>
                <Text
                  style={
                    Object {
                      "color": "#3E3C37",
                      "fontSize": 12,
                    }
                  }
                >
                  ISBN13 -
                   
                  9780451524935
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3E3C37",
                      "fontSize": 12,
                    }
                  }
                >
                  ISBN10 -
                   
                  0451524934
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginVertical": 10,
                "opacity": 1,
              }
            }
            testID="bookDetailButton"
          >
            <Image
              source={
                Object {
                  "uri": "smallThumbnail2",
                }
              }
              style={
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderTopLeftRadius": 10,
                  "height": 200,
                  "overflow": "hidden",
                  "width": 125,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 175,
                  "justifyContent": "space-between",
                  "padding": 10,
                  "width": 210,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "maxHeight": 75,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "width": 175,
                      }
                    }
                  >
                    1984
                  </Text>
                  <Text>
                    EN
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                    }
                  }
                >
                  George Orwell
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                    }
                  }
                >
                  Another author
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                    }
                  }
                >
                  ...
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3E3C37",
                      "fontSize": 12,
                    }
                  }
                >
                  No publisher found
                </Text>
              </View>
              <View>
                <Text
                  style={
                    Object {
                      "color": "#3E3C37",
                      "fontSize": 12,
                    }
                  }
                >
                  ISBN13 -
                   
                  9780451524935
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3E3C37",
                      "fontSize": 12,
                    }
                  }
                >
                  ISBN10 -
                   
                  0451524934
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "height": 130,
          "width": "100%",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;

exports[`Given a BookResults component When it is rendered And there are no book results searching by author and publisher Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        Author: AUTHOR
      </Text>
      <Text
        style={
          Object {
            "color": "#3E3C37",
            "fontSize": 12,
            "paddingHorizontal": 5,
          }
        }
      >
        Publisher: PUBLISHER
      </Text>
    </View>
    <Text>
      0 results
    </Text>
    <View
      style={
        Object {
          "height": 130,
          "width": "100%",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
